File: 55/Solution.java
class Solution {
    public boolean canJump(int[] nums) {
        int rightMost = 0;
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (i <= rightMost) {
                rightMost = Math.max(rightMost, i + nums[i]);
            } else {
                return false;
            }
            if (rightMost >= n - 1) {
                return true;
            }
        }
        return true;
    }
}

File: 55/solution.go
package _55

func canJump(nums []int) bool {
	rightMost := 0
	n := len(nums)
	for i := 0; i < n; i++ {
		if i <= rightMost {
			rightMost = max(rightMost, i+nums[i])
		} else {
			return false
		}
		if rightMost >= n-1 {
			return true
		}
	}
	return true
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
